.github/workflows/build-ami.yml

```diff
@@ -56,13 +56,13 @@ jobs:
           
           - name: Packer Validate
             id: validate
-            run: packer validate -var "optimum_version=${{ github.event.inputs.tag }}" -var "region=${{ env.AWS_REGION }}" hcl2-files 
+            run: packer validate -var "optimum_version=${{ github.event.inputs.tag || github.event.repository.default_branch }}" -var "region=${{ env.AWS_REGION }}" hcl2-files 
```

Comment:
Why is it here `github.event.repository.default_branch` and not `github.ref`? Wouldn't this mean that when we open a PR with changes, we would still install the main branch? 

Lets align this. Otherwise i can be confusing for others when the need to makes changes for the ami and the main branch.

Reply:
Yes, it would then install the main branch. 

The workflow either gets triggered when someone makes changes to `infrastructure/ami/**'`  then in that case the `optimum_version` would be `main` and if it is triggered automatically using scheduler or manually through cli then it would either be `main` or the `tag` passed. 

If we want to build this image at any particular change in the current repo, which makes sense then I need to remove the path: `infrastructure/ami/**`

Reply:
okay let's keep it but add a comment that others understand why.

Reply:
@philschmid Added the comments

Reply:
So shall I go ahead and merge it?

Reply:
Yes feel free to merge